prefix.go

```diff
@@ -217,23 +238,34 @@ func (i *ipamer) acquireChildPrefixInternal(parentCidr string, length uint8) (*P
 		ipset.RemovePrefix(cpipprefix)
 	}
 
-	cp, _, ok := ipset.IPSet().RemoveFreePrefix(length)
-	if !ok {
-		pfxs := ipset.IPSet().Prefixes()
-		if len(pfxs) == 0 {
-			return nil, fmt.Errorf("no prefix found in %s with length:%d", parentCidr, length)
-		}
+	var cp netaddr.IPPrefix
 
-		var availablePrefixes []string
-		for _, p := range pfxs {
-			availablePrefixes = append(availablePrefixes, p.String())
-		}
-		adj := "are"
-		if len(availablePrefixes) == 1 {
-			adj = "is"
+	if specificChildRequest {
```

Comment:
happy case first please

Reply:
Are you referring to the if statement checking child prefix is contained in the ipset of the parent?

Reply:
this case should be the else block